src/components/grid/grid.pw.tsx

```diff
@@ -0,0 +1,533 @@
+        break;
+      case "small":
+        screenDimension = { width: 959, height: 900 };
+        break;
+      case "medium":
+        screenDimension = { width: 1259, height: 900 };
+        break;
+      case "large":
+        screenDimension = { width: 1920, height: 900 };
+        break;
+      case "extra large":
+        screenDimension = { width: 1922, height: 900 };
+        break;
+      default:
+        throw new Error(
+          `${screenSize} is not defined in a scope. We want to resize to only for types of viewport`
+        );
+    }
+
+    test(`when viewport size is ${screenSize}, grid container has correct padding and row gap size`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SimpleGridExample />);
+      await page.setViewportSize(screenDimension);
+
+      await expect(gridContainer(page)).toHaveCSS(
+        "padding-left",
+        `${padding}px`
+      );
+      await expect(gridContainer(page)).toHaveCSS("row-gap", `${gridGap}px`);
+    });
+  });
+
+  ["start", "end", "center", "stretch"].forEach((alignment) => {
+    test(`grid item correctly has alignment set to ${alignment}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SimpleGridExample alignSelf={alignment} />);
+
+      await expect(gridItem(page, 0)).toHaveCSS("align-self", alignment);
+    });
+  });
+
+  ["start", "end", "center", "stretch"].forEach((justification) => {
+    test(`grid item correctly has justification set to ${justification}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SimpleGridExample justifySelf={justification} />);
+
+      await expect(gridItem(page, 0)).toHaveCSS("justify-self", justification);
+    });
+  });
+
+  test("grid item has correct start and end columns when gridColumn prop is passed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SimpleGridExample gridColumn="4 / 10" />);
+
+    await expect(gridItem(page, 0)).toHaveCSS("grid-column-start", "4");
+    await expect(gridItem(page, 0)).toHaveCSS("grid-column-end", "10");
+  });
+
+  test("grid item has correct start and end rows when gridRow prop is passed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SimpleGridExample gridRow="4 / 11" />);
+
+    await expect(gridItem(page, 0)).toHaveCSS("grid-row-start", "4");
+    await expect(gridItem(page, 0)).toHaveCSS("grid-row-end", "11");
+  });
+
+  test("grid item has correct start and end columns and rows when gridArea prop is passed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SimpleGridExample gridArea="3 / 4 / 11 / 10" />);
+
+    await expect(gridItem(page, 0)).toHaveCSS("grid-column", "4 / 10");
+    await expect(gridItem(page, 0)).toHaveCSS("grid-row", "3 / 11");
+  });
+
+  test("correctly renders grid items when each have responsive settings passed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <GridContainer>
+        <GridItem
```

Comment:
It might make tests easier to read if this mount was moved to `components.test-pw.tsx`. It could also be reused for the accessibility test below.